src/odemis/acq/move.py

```diff
@@ -390,11 +392,21 @@ def _doCryoSwitchSamplePosition(future, stage, align, target):
             if current_label == LOADING:
                 # As moving from loading position requires re-referencing the stage, move all axes together to
                 # prevent invalid/reachable position error
-                sub_moves.append((stage, filter_dict({'x', 'y', 'z'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'y'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'z'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'rx', 'rz'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'x'}, target_pos[target])))
             else:
+                sub_moves.append((stage, filter_dict({'x'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'y'}, target_pos[target])))
                 sub_moves.append((stage, filter_dict({'z'}, target_pos[target])))
-                sub_moves.append((stage, filter_dict({'x', 'y'}, target_pos[target])))
-            sub_moves.append((stage, filter_dict({'rx', 'rz'}, target_pos[target])))
+                sub_moves.append((stage, filter_dict({'rx', 'rz'}, target_pos[target])))
```

Comment:
Here, it's not clear why it was changed. Is the idea just that each movement should be separate? Then, as each move is small (and sort of undefined), we can just do them in alphabetical order? The rotation would be last, because it's assumed that previous it's 0, so it's safer to rotate last (as the pivot point is under the pole piece). 

Or should we check that and do the rotation earlier if the current rx > target rx? Shouldn't be X moved last, as it's most likely to get closer to the pole-piece, so we should do it only when the rest is in place.